### PR TITLE
create course with subject

### PR DESCRIPTION
**Pull request by BuiDuySon**

### CODE REVIEW

File: app/controllers/courses_controller.rb [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    redirect_to courses_url
+  end
+
+  private
+  def set_course
+    @course = Course.find(params[:id])
+  end
+
+  def course_params
+    params.require(:course).permit(:name, :description, :start, :finish, { :subject_ids => [] })
```
Comment (dieunb) [inline app/controllers/courses_controller.rb]: `{:subject_ids => []}`

File: app/models/course.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+class Course < ActiveRecord::Base
+  has_many :courses_subjects
+  has_many :subjects, :through => :courses_subjects
```
Comment (dieunb) [inline app/models/course.rb]: `through: :courses_subjects`

File: app/models/subject.rb [hunk 3]
```diff
@@ -0,0 +1,9 @@
+class Subject < ActiveRecord::Base
+  belongs_to :user
+  has_many :courses_subjects
+  has_many :courses, :through => :courses_subjects
```
Comment (dieunb) [inline app/models/subject.rb]: `through: :courses_subjects`

File: app/views/courses/show.html.erb [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+  <strong>Finish:</strong>
+  <%= @course.finish %>
+</p>
+
+<h2>
+  <strong>List subject of this course</strong>
+</h2>
+
+<% @subjects.each do |subject| %>
+<%= subject.name %> : <%= subject.description %><br>
```
Comment (dieunb) [inline app/views/courses/show.html.erb]: indent

File: app/controllers/courses_controller.rb [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+
+  def new
+    @course = Course.new
+  end
+
+  def edit
+  end
+
+  def create
+    @course = Course.new(course_params)
```
Comment (chuanhtuan) [inline app/controllers/courses_controller.rb]: ()

File: app/controllers/courses_controller.rb [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+    if @course.save
+      redirect_to @course
+    else
+      render :new
+    end
+  end
+
+  def update
+    if @course.update(course_params)
```
Comment (chuanhtuan) [inline app/controllers/courses_controller.rb]: use update_attributes instead of update method

File: app/controllers/courses_controller.rb [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+  end
+
+  def destroy
+    @course.destroy
+    redirect_to courses_url
+  end
+
+  private
+  def set_course
+    @course = Course.find(params[:id])
```
Comment (chuanhtuan) [inline app/controllers/courses_controller.rb]: ()

File: app/models/user.rb [hunk 8]
```diff
@@ -11,6 +11,7 @@ class User < ActiveRecord::Base
   validate :file_size => {maximum: 1.megabytes.to_i}
 
   has_secure_password
+  scope :trainee, -> {where(role: 'trainee')}
```
Comment (dieunb) [inline app/models/user.rb]: `trainees` plural not `trainee` single

File: app/controllers/courses_controller.rb [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+  def destroy
+  end
+
+  private
+  def set_course
+    @course = Course.find params[:id]
+  end
+
+  def course_params
+    params.require(:course).permit(:name, :description, :start, :finish, {:subject_ids => []})
```
Comment (dieunb) [inline app/controllers/courses_controller.rb]: `subject_ids: []`

File: app/controllers/suppervisor/courses_controller.rb [hunk 10]
```diff
@@ -0,0 +1,53 @@
...
+    redirect_to suppervisor_courses_path
+  end
+
+  private
+  def set_course
+    @course = Course.find params[:id]
+  end
+
+  def course_params
+    params.require(:course).permit(:name, :description, :start, :finish, {:subject_ids => []})
```
Comment (dieunb) [inline app/controllers/suppervisor/courses_controller.rb]: `subject_ids: []`

File: app/controllers/users_controller.rb [hunk 11]
```diff
@@ -3,7 +3,7 @@ class UsersController < ApplicationController
   before_action :check_right_user, only: [ :edit, :update ]
```
Comment (dieunb) [inline app/controllers/users_controller.rb]: `[:edit, :update]`

File: app/models/user.rb [hunk 12]
```diff
@@ -11,6 +11,7 @@ class User < ActiveRecord::Base
   validate :file_size => {maximum: 1.megabytes.to_i}
```
Comment (dieunb) [inline app/models/user.rb]: `file_size: {...}`

File: app/views/suppervisor/courses/index.html.erb [hunk 13]
```diff
@@ -0,0 +1,31 @@
...
+  <tbody>
+    <% @courses.each do |course| %>
+      <tr>
+        <td><%= course.name %></td>
+        <td><%= course.description %></td>
+        <td><%= course.start %></td>
+        <td><%= course.finish %></td>
+        <td><%= link_to 'Show', suppervisor_course_path(course) %></td>
+        <td><%= link_to 'Edit', edit_suppervisor_course_path(course) %></td>
+        <td><%= link_to 'Destroy', suppervisor_course_path(course), method: :delete, data: { confirm: 'Are you sure?' } %></td>
```
Comment (dieunb) [inline app/views/suppervisor/courses/index.html.erb]: line length is too long

File: app/models/courses_subject.rb [hunk 14]
```diff
@@ -0,0 +1,8 @@
+class CoursesSubject < ActiveRecord::Base
+  belongs_to :course
+  belongs_to :subject
+  
+  def self.getCourseSubject(course,subject)
```
Comment (dieunb) [inline app/models/courses_subject.rb]: missing a space `(course,subject)`

**Pull request merged**